engine/xml-config/src/main/java/org/datacleaner/job/JaxbJobReader.java

```diff
@@ -915,9 +919,50 @@ private void applyProperties(final ComponentBuilder builder,
                     final Converter<?> customConverter = configuredProperty.createCustomConverter();
                     final Object value = stringConverter.deserialize(stringValue, configuredProperty.getType(),
                             customConverter);
+               
+                    if (name.equals(CoalesceMultipleFieldsTransformer.PROPERTY_UNITS)) {
+                        /* This part of the code refers to the situation when we
+                         open the job as a template where we need to replace
+                         the name of the columns with the mapped ones */
+                        final CoalesceUnit[] units = (CoalesceUnit[]) value;
+
+                        final ArrayList<CoalesceUnit> newUnitsList = new ArrayList<CoalesceUnit>();
+                        final Set<Entry<String, InputColumn<?>>> mappingColumnsSet = mappingInputColumns.entrySet();
+                        for (int i = 0; i < units.length; i++) {
+                            final String[] oldInputColumnNames = units[i].getInputColumnNames();
+                            final ArrayList<InputColumn<?>> newInputColumns = new ArrayList<InputColumn<?>>();
+                            for (Entry<String, InputColumn<?>> entry : mappingColumnsSet) {
+                                final String column_id = entry.getKey();
+                                final String path = getPath(columnsTypes, column_id);
```

Comment:
Does this handle the case where the column isn't an input column, but the output from a transformer?